casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -96,6 +107,32 @@ object BlockCreator {
         isActive = parents
           .traverse(b => isActiveValidator(b, runtimeManager, validatorIdentity))
           .map(_.forall(identity))
+        _ <- Log[F].info(s"Check isActive:${isActive} for creating block")
+
+        invalidLatestMessagesExcludeUnbonded <- invalidLatestMessages.toList.filterA(
+                                                 v =>
+                                                   parents
+                                                     .traverse(
+                                                       b =>
+                                                         isBonded(
+                                                           b,
+                                                           runtimeManager,
+                                                           PublicKey(v._1)
+                                                         )
+                                                     )
+                                                     .map(_.forall(identity))
+                                               )
```

Comment:
We have parent blocks here and the bonds list is already in the block (and validated with replay) so after all we don't need to execute Rholang to get it.